// ngx-strongly-typed-forms/ngx-strongly-typed-forms.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/forms'), require('@angular/core')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/forms', '@angular/core'], factory) :
	(factory((global['ngx-strongly-typed-forms'] = {}),global.forms,global.core));
}(this, (function (exports,forms,core) { 'use strict';

const AbstractControl$1 = forms.AbstractControl;
const FormArray$1 = forms.FormArray;
const FormControl$1 = forms.FormControl;
const FormGroup$1 = forms.FormGroup;
const FormBuilder$1 =  forms.FormBuilder;

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var NgxStronglyTypedFormsModule = /** @class */ (function () {
    function NgxStronglyTypedFormsModule() {
    }
    NgxStronglyTypedFormsModule.decorators = [
        { type: core.NgModule, args: [{
                    providers: [FormBuilder$1],
                },] },
    ];
    return NgxStronglyTypedFormsModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * @template T
 * @param {?} control
 * @return {?}
 */
function toUntyped(control) {
    return /** @type {?} */ (control);
}

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

exports.AbstractControl = AbstractControl$1;
exports.FormArray = FormArray$1;
exports.FormGroup = FormGroup$1;
exports.FormControl = FormControl$1;
exports.FormBuilder = FormBuilder$1;
exports.NgxStronglyTypedFormsModule = NgxStronglyTypedFormsModule;
exports.toUntyped = toUntyped;

Object.defineProperty(exports, '__esModule', { value: true });

})));
