// ngx-strongly-typed-forms/public_api.js
import { AbstractControl, FormArray, FormBuilder, FormControl, FormGroup } from '@angular/forms';
import { NgModule } from '@angular/core';

const AbstractControl$1 = AbstractControl;
const FormArray$1 = FormArray;
const FormControl$1 = FormControl;
const FormGroup$1 = FormGroup;
const FormBuilder$1 =  FormBuilder;

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var NgxStronglyTypedFormsModule = /** @class */ (function () {
    function NgxStronglyTypedFormsModule() {
    }
    NgxStronglyTypedFormsModule.decorators = [
        { type: NgModule, args: [{
                    providers: [FormBuilder$1],
                },] },
    ];
    return NgxStronglyTypedFormsModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * @template T
 * @param {?} control
 * @return {?}
 */
function toUntyped(control) {
    return /** @type {?} */ (control);
}

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { AbstractControl$1 as AbstractControl, FormArray$1 as FormArray, FormGroup$1 as FormGroup, FormControl$1 as FormControl, FormBuilder$1 as FormBuilder, NgxStronglyTypedFormsModule, toUntyped };
